src/embedding.rs

```diff
@@ -191,6 +234,58 @@ impl MatrixWrapper for MMapMatrix {
         }
     }
 
+    fn init_with_eigenvectors(
+        rows: usize,
+        cols: usize,
+        sparse_matrix_reader: Arc<SparseMatrix>,
+        order: TruncatedOrder,
+    ) -> Self {
+        let uuid = Uuid::new_v4();
+        let file_name = format!("{}_matrix_{}", sparse_matrix_reader.get_id(), uuid);
+        let mut mmap = create_mmap(rows, cols, file_name.as_str());
+
+        let mut discrete_laplacian = Array2::zeros((rows, rows));
+
+        for entry in &sparse_matrix_reader.entries {
```

Comment:
Some laplacian code probably could be extracted and reused inboth modes (mmap, in-memory). There is copy-pasta here